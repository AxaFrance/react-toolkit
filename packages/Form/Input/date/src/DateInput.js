import React from 'react';
import PropTypes from 'prop-types';

import {
  Field,
  HelpMessage,
  FieldInput,
  FieldConstants,
  FormClassManager,
} from '@axa-fr/react-toolkit-form-core';
import { InputManager } from '@axa-fr/react-toolkit-core';
import CustomDate from './CustomDate';

const propTypes = {
  ...FieldConstants.propTypes,
  value: PropTypes.object,
  viewValue: PropTypes.string,
  locale: PropTypes.string,
  format: PropTypes.string,
};

const defaultProps = {
  ...FieldConstants.defaultProps,
  value: null,
  viewValue: null,
  locale: 'fr',
  format: 'dd/MM/yyyy',
};

const DateInput = (props) => {
  const {
    classModifier,
    message,
    children,
    helpMessage,
    id,
    disabled,
    classNameContainerLabel,
    classNameContainerInput,
    label,
    messageType,
    placeholder,
    isVisible,
    forceDisplayMessage,
    className,
    name,
    value,
    locale,
    format,
    viewValue,
    onChange,
    readOnly,
    ...otherProps
  } = props;

  if (!isVisible) {
    return null;
  }
  const inputClassModifier = FormClassManager.getInputClassModifier(
    classModifier,
    children
  );
  const inputFieldClassModifier = FormClassManager.getFieldInputClassModifier(
    classModifier,
    disabled
  );
  const inputId = InputManager.getInputId(id);
  return (
    <Field
      label={label}
      id={inputId}
      message={message}
      messageType={messageType}
      placeholder={placeholder}
      isVisible={isVisible}
      forceDisplayMessage={forceDisplayMessage}
      className={className}
      classModifier={classModifier}
      classNameContainerLabel={classNameContainerLabel}
      classNameContainerInput={classNameContainerInput}>
      <FieldInput
        helpMessage={helpMessage}
        message={message}
        messageType={messageType}
        className="af-datepicker__container"
        classModifier={inputFieldClassModifier}>
        <CustomDate
          name={name}
          id={inputId}
          value={value}
          locale={locale}
          format={format}
          viewValue={viewValue}
          onChange={onChange}
          readOnly={readOnly}
          disabled={disabled}
          placeholder={placeholder}
          classModifier={inputClassModifier}
          {...otherProps}
        />
        <span className="glyphicon glyphicon-calendar" />
        {children}
      </FieldInput>
      <HelpMessage message={helpMessage} isVisible={!message} />
    </Field>
  );
};

DateInput.propTypes = propTypes;
DateInput.defaultProps = defaultProps;

export default DateInput;
