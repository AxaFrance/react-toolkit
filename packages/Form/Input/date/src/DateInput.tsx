import React, {
  ComponentPropsWithoutRef,
  ComponentPropsWithRef,
  ReactNode,
} from 'react';

import {
  Field,
  HelpMessage,
  FieldInput,
  useInputClassModifier,
} from '@axa-fr/react-toolkit-form-core';
import { useId } from '@axa-fr/react-toolkit-core';
import Date from './CustomDate';

type Props = Omit<
  ComponentPropsWithoutRef<typeof Date>,
  'placeholderText'
> &
  ComponentPropsWithoutRef<typeof Field> & {
    placeholder?: string;
    helpMessage?: ReactNode;
    children?: ReactNode;
  };

const DateInput = ({
  classModifier,
  message,
  children,
  helpMessage,
  id,
  classNameContainerLabel,
  classNameContainerInput,
  label,
  messageType,
  isVisible,
  forceDisplayMessage,
  className,
  name,
  value,
  locale,
  onChange,
  readOnly,
  disabled,
  ...otherProps
}: Props) => {
  const inputId = useId(id);
  const { inputClassModifier, inputFieldClassModifier } = useInputClassModifier(
    classModifier,
    disabled,
    !!children
  );
  return (
    <Field
      label={label}
      id={inputId}
      message={message}
      messageType={messageType}
      isVisible={isVisible}
      forceDisplayMessage={forceDisplayMessage}
      className={className}
      classModifier={classModifier}
      classNameContainerLabel={classNameContainerLabel}
      classNameContainerInput={classNameContainerInput}>
      <FieldInput
        className="af-form__date"
        classModifier={inputFieldClassModifier}>
        <Date
          name={name}
          id={inputId}
          value={value}
          classModifier={inputClassModifier}
          disabled={disabled}
          {...otherProps}
        />
        {children}
      </FieldInput>
      <HelpMessage message={helpMessage} isVisible={!message} />
    </Field>
  );
};

export default DateInput;
