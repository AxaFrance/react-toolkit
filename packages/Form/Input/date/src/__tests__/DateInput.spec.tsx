import { render } from '@testing-library/react';
import React from 'react';
import DateInput from '../DateInput';

describe('<NumberInput>', () => {
  it('renders DateInput correctly', () => {
    const { asFragment, getByLabelText } = render(
      <DateInput
        label="Date début"
        id="iddateinput"
        name="namedateinput"
        value={new Date('2024-03-04')}
        onChange={() => {}}
      />
    );

    expect(asFragment()).toMatchSnapshot();

    const inputElement = getByLabelText('Date début');
    expect(inputElement).toHaveAttribute('type', 'date');
    expect(inputElement).toHaveValue('2024-03-04');
    expect(inputElement).toHaveClass('af-form__input-date');

    expect(inputElement.id).toBe('iddateinput');
  });
});
