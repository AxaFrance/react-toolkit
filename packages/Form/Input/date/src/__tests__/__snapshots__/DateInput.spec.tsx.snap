// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateInput> renders DateInput correctly 1`] = `
<DocumentFragment>
  <div
    class="row af-form__group"
  >
    <div
      class="col-md-2"
    >
      <label
        class="af-form__group-label"
        for="iddateinput"
      >
        date de naissance
      </label>
    </div>
    <div
      class="col-md-10"
    >
      <div
        class="af-datepicker__container"
      >
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class="af-datepicker"
              id="iddateinput"
              name="placeImage"
              type="text"
              value=""
            />
          </div>
        </div>
        <span
          class="glyphicon glyphicon-calendar"
        />
      </div>
      <small
        class="af-form__help"
      />
    </div>
  </div>
</DocumentFragment>
`;
